assets/system/game_modules/kami.mjs

```diff
@@ -30,34 +30,87 @@
  *  POSSIBILITY OF SUCH DAMAGE.
 **/
 
-import from from 'from';
-
 export default new
 class Kami
 {
-	constructor(title = Sphere.Game.name)
+	constructor()
 	{
 		this.enabled = SSj.now() > 0;
-		this.title = title;
 		this.records = [];
+		this.outputBordered = false;
+		this.placeholder = new Record();
 		if (this.enabled)
 			this.exitJob = Dispatch.onExit(() => this.finish());
 	}
 
 	finish()
 	{
-		SSj.log(`Profiling has completed for "${this.title}"`);
-		let sortedRecords = from(this.records)
-			.where(it => it.count > 0)
-			.descending(it => it.totalTime / it.count);
-		for (const record of sortedRecords) {
-			let averageTime = Math.round(record.totalTime / record.count / 1000).toLocaleString();
-			let count = record.count.toLocaleString();
-			let totalTime = Math.round(record.totalTime / 1000).toLocaleString();
-			SSj.log(`  ${count} occurrences "${record.description}" took ${totalTime} mcs (avg. ${averageTime} mcs)`);
-			record.target[record.methodName] = record.originalFunction;
-		}
+		if (!this.enabled)
+			return;
+		
 		this.exitJob.cancel();
+
+		let totalTime = 0;
+		let totalAverage = 0;
+		for (let i = 0; i < this.records.length; ++i) {
+			if (this.records[i].count === 0) {
+				this.records.splice(i, 1);
+				--i;
+			}
+			else {
+				const record = this.records[i];
+				record.averageTime = record.totalTime / record.count;
+				totalTime += record.totalTime;
+				totalAverage += record.averageTime;
+			}
+		}
+
+		this.records.sort(function(a, b) { return b.averageTime - a.averageTime; });
+
+		let consoleOutput = [
+			[ "Event" ],
+			[ "Calls" ],
+			[ "Time (\u03bcs)" ],
+			[ "% Total" ],
+			[ "Avg (\u03bcs)" ],
+			[ "% Avg" ] ];
+		for (const record of this.records) {
+			consoleOutput[0].push(record.description);
+			consoleOutput[1].push(record.count.toLocaleString());
+			consoleOutput[2].push(Math.round(record.totalTime / 1000).toLocaleString());
+			consoleOutput[3].push(Math.round(100 * record.totalTime / totalTime) + "%");
+			consoleOutput[4].push(Math.round(record.averageTime / 1000).toLocaleString());
+			consoleOutput[5].push(Math.round(100 * record.averageTime / totalAverage) + "%");
+			if (record.attached === true)
```

Comment:
Explicit comparison against `true` is redundant.  I know this is your style but it goes against the usual conventions for miniRT modules. 😛

Reply:
Not actually redundant as it stops the engine from doing toBoolean() on the variable.

I’ll change it but IMO should always make this explicit to limit chance of weird errors.

Reply:
Yeah, I saw the strict-equal and am aware of the implicit conversion, I just don't like the extra visual noise it adds when reading the code.

Reply:
I generally prioritise function over appearance (though try to ensure everything is nonetheless clear and easy to read/follow - or add comments...) I guess my code is generally a lot longer than yours for the same thing overall.